### PR TITLE
[COMMON] Enable support for external modems and SM8250

### PR DESCRIPTION
**Pull request by kholk**
This patchset enables support for the external modems and the SM8250 SoC.

### CODE REVIEW

File: hardware/tftp/Android.mk [hunk 1]
```diff
@@ -1,6 +1,6 @@
 LOCAL_PATH := $(call my-dir)
 
-ifneq ($(filter sdm660 msm8998 sdm845 sm6125 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter sdm660 msm8998 sdm845 sm6125 sm8150 sm8250,$(TARGET_BOARD_PLATFORM)),)
```
Comment (MarijnS95) [inline hardware/tftp/Android.mk:3]:
```suggestion
ifneq ($(filter sdm660 msm8998 sdm845 $(TRINKET) sm8150 $(KONA),$(TARGET_BOARD_PLATFORM)),)
```

Let's stick with and eat our own dogfood.

File: hardware/tftp/Android.mk [hunk 2]
```diff
@@ -18,25 +18,41 @@ target_prefixes := \
     slpi \
     cdsp
 
+target_soc := \
+    msm
+
+ifeq ($(TARGET_USES_ESOC),true)
+target_soc += \
+    mdm
+endif
```
Comment (MarijnS95) [inline hardware/tftp/Android.mk:27]: You know by now I'm a man of useless and unreasonable requests:

```suggestion
target_soc := \
    msm \
    $(if $(filter true,$(TARGET_USES_ESOC)),mdm,)
```

File: hardware/tftp/Android.mk (hunk 1 above)
Comment (MarijnS95) [reply hardware/tftp/Android.mk:3]: Anyway, why are we still carrying this completely useless and annoying guard? All platforms supported on R branches need these symlinks...

File: hardware/tftp/Android.mk [hunk 3]
```diff
@@ -18,25 +18,41 @@ target_prefixes := \
...
-    ) \
-)
+SONY_SYMLINKS := \
+    $(foreach tgtsoc,$(target_soc), \
+        $(foreach prefix,$(target_prefixes), \
+            $(foreach s,$(target_combinations), \
+                $(eval p := $(subst :,$(space),$(s))) \
+                $(word 1,$(p)):$(TARGET_COPY_OUT_VENDOR)/rfs/$(tgtsoc)/$(prefix)/$(word 2,$(p)) \
+            ) \
+        )
```
Comment (MarijnS95) [inline hardware/tftp/Android.mk:37]:
```suggestion
        ) \
```

```
device/sony/common/hardware/tftp/Android.mk:30: error: unterminated call to function 'foreach': missing ')'.
```

**Review by MarijnS95 — CHANGES_REQUESTED** [review]

File: hardware/tftp/Android.mk [hunk 4]
```diff
@@ -18,25 +18,41 @@ target_prefixes := \
...
 
 # Edgecase for readwrite folders that all point to their own persist folder:
-SONY_SYMLINKS += $(foreach prefix,$(target_prefixes), \
-    /mnt/vendor/persist/rfs/msm/$(prefix):$(TARGET_COPY_OUT_VENDOR)/rfs/msm/$(prefix)/readwrite \
-)
+SONY_SYMLINKS += \
+    $(foreach tgtsoc,$(target_soc), \
+        $(foreach prefix,$(target_prefixes), \
+            /mnt/vendor/persist/rfs/msm/$(prefix):$(TARGET_COPY_OUT_VENDOR)/rfs/msm/$(prefix)/readwrite \
+        )
```
Comment (MarijnS95) [inline hardware/tftp/Android.mk:45]:
```suggestion
        ) \
```

And here :)

File: hardware/tftp/Android.mk [hunk 5]
```diff
@@ -18,25 +18,41 @@ target_prefixes := \
...
-    /data/vendor/tombstones/modem:$(TARGET_COPY_OUT_VENDOR)/rfs/msm/mpss/ramdumps \
-    /data/vendor/tombstones/lpass:$(TARGET_COPY_OUT_VENDOR)/rfs/msm/adsp/ramdumps \
-    /data/vendor/tombstones/rfs/cdsp:$(TARGET_COPY_OUT_VENDOR)/rfs/msm/cdsp/ramdumps \
-    /data/vendor/tombstones/rfs/slpi:$(TARGET_COPY_OUT_VENDOR)/rfs/msm/slpi/ramdumps
+    $(foreach tgtsoc,$(target_soc), \
+        /data/vendor/tombstones/modem:$(TARGET_COPY_OUT_VENDOR)/rfs/$(tgtsoc)/mpss/ramdumps \
+        /data/vendor/tombstones/lpass:$(TARGET_COPY_OUT_VENDOR)/rfs/$(tgtsoc)/adsp/ramdumps \
+        /data/vendor/tombstones/rfs/cdsp:$(TARGET_COPY_OUT_VENDOR)/rfs/$(tgtsoc)/cdsp/ramdumps \
+        /data/vendor/tombstones/rfs/slpi:$(TARGET_COPY_OUT_VENDOR)/rfs/$(tgtsoc)/slpi/ramdumps
+    )
```
Comment (MarijnS95) [inline hardware/tftp/Android.mk:55]:
```suggestion
        /data/vendor/tombstones/rfs/slpi:$(TARGET_COPY_OUT_VENDOR)/rfs/$(tgtsoc)/slpi/ramdumps \
    )
```

And here.

File: hardware/tftp/Android.mk (hunk 2 above)
Comment (kholk) [reply hardware/tftp/Android.mk:27]: That's ugly and it's a big no for me.

File: hardware/tftp/Android.mk (hunk 4 above)
Comment (kholk) [reply hardware/tftp/Android.mk]: All of the missed '\' have been solved yesterday already....

File: hardware/tftp/Android.mk (hunk 1 above)
Comment (kholk) [reply hardware/tftp/Android.mk:3]: Unnecessary style change. Just go for it as it is.

Perhaps one day we may just remove the barriers entirely, as they may make no sense anymore, like you suggested.

**Comment by kholk** [comment]
This PR should be ready to merge, unless anybody else has comments.

File: hardware/tftp/Android.mk (hunk 4 above)
Comment (MarijnS95) [reply hardware/tftp/Android.mk]: Fun fact, these suggestions were posted before that!

But for some reason I am not allowed to mark _my own comments_ as resolved. Send the blame to GH!

File: hardware/tftp/Android.mk (hunk 1 above)
Comment (MarijnS95) [reply hardware/tftp/Android.mk:3]: Let's remove them at some point...

**Review by MarijnS95 — APPROVED** [review]
No comments, merge it please!

File: hardware/tftp/Android.mk (hunk 2 above)
Comment (MarijnS95) [reply hardware/tftp/Android.mk:27]: That's what I said :) My itch prefers conciseness over ugliness :D

**Pull request merged**